### PR TITLE
Uber Eats clone using the meals_db and the involvement api - Capstone JS module 2

### PR DESCRIPTION
**Pull request by dennis-every**
This is the pull request to merge the development branch into the main branch as per the git-flow workflow. For this project, we built an Uber Eats clone using the meals DB and the involvement APIs. We built the app with the following features as required per the assignment:
- A home page that loads the meals images and text from the meals DB API
- On the home page the likes for each meal are loaded from the involvement API
- A functionality to allow users to like a meal only once on each page load
- A functionality to open a modal when the user clicks on the comments button of a meal
- Each meal pop-up modal display an image of the meal and relevant information about this meal
- The modal also shows the comments received for each meal if any
- User can create a comment on a modal of a specific meal
- The modal updates the modal page after the user has submitted the comment
- The app posts the comments to the involvement API
- We developed relevant tests for counting meals and comments 
- The app has been deployed to [Gitub pages](https://dennis-every.github.io/dd_eats/dist/)

### CODE REVIEW

File: src/modules/comments/commentsCounter.test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    <li>Comment</li>
+    <li>Comment</li>
+    <li>Comment</li>
+    <li>Comment</li>
+    <li>Comment</li>
+  `;
+    commentsCounter();
+    expect(title.innerHTML).toStrictEqual('Comments (7)');
+  });
+});
```
Comment (ShadyShawkat) [inline src/modules/comments/commentsCounter.test.js:38]: - [ ] Kindly make sure that all tests pass successfully.
image

File: src/modules/meals/counter.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export const countMeals = (meals) => {
+  const mealsCount = meals ? meals.length : 0;
+  return mealsCount;
+};
```
Comment (ShadyShawkat) [inline src/modules/meals/counter.js:4]: - [ ] Please make sure that this method is implemented using [this](https://github.com/dennis-every/dd_eats/issues/6#:~:text=A%20counter%20function%20should%20look%20for%20specific%20DOM%20elements%20(e.g.%20for%20the%20comments%20counter%20it%20should%20look%20for%20comments)%20and%20make%20the%20counting%20based%20on%20what%20is%20actually%20displayed%20on%20the%20page.) description which is by counting the DOM elements and not by using the fetched items from the API directly. (Also please make sure to also account for the tests)

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi team 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done on making the required changes from the previous reviewer. All good now 💯 

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**